homeassistant/components/switch/kankun.py

```diff
@@ -0,0 +1,101 @@
+"""
+Support for customised Kankun SP3 wifi switch.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.kankun/
+"""
+import logging
+import requests
+
+from homeassistant.components.switch import SwitchDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+    """Find and return kankun switches."""
+    switches = config.get('switches', {})
+    devices = []
+
+    for dev_name, properties in switches.items():
+        devices.append(
+            KankunSwitch(
+                hass,
+                properties.get('name', dev_name),
```

Comment:
Please add a voluptuous schema validation that also includes the default values.